File: Entities/Response/RouteResponse.cs
using Entities.Models;


namespace Entities.Response
{
    public class RouteResponse
    {
        public string Id { get; set; } = string.Empty;
        public string PilotId { get; set; } = string.Empty;
        public string Observations { get; set; } = string.Empty;
        public int State { get; set; } = 1;
        public string CreatedAt { get; set; } = string.Empty;
        public string? UpdatedAt { get; set; }  
        public string CreatedBy { get; set; } = string.Empty;
        public string? UpdatedBy { get; set; }


        public virtual Pilot? Pilot { get; set; }
    }
}

File: dashboard_react.client/src/types/RouteResponse.ts
import { PilotResponse } from "./PilotResponse";
import { RouteDetailsResponse } from "./RouteDetailsResponse";

export interface RouteResponse {
  id: string;
  pilotId: string;
  observations: string;
  state: number;
  createdAt: string | null;
  updatedAt: string | null;
  createdBy: string | null;
  updatedBy: string | null;
  pilot: PilotResponse;
  routeDetails: RouteDetailsResponse[];
}
